### DISCUSSION TITLE
feature request: set service timeout globally

### DISCUSSION THREAD

**Discussion by danielholden** [Kong/kong#7285] [Ideas and feature requests]
### Summary

At the moment the global timeout of an upstream service is 60 seconds, it would be nice to be able to globally configure this.

At the moment the only way to do this is set the `connect_timeout` attribute for each service.

**Comment by thibaultcha** [comment]
Agreed.

I just linked to a tangentially related conversation describing the need for global configuration properties for the upstream keepalive pool(s). We'll squeeze these new global properties as soon as possible, and most likely when implementing the dynamic upstream keepalive pools in Kong, after bumping to OpenResty 1.17.x.1, which most likely means Kong 2.x, in the next few months.

**Comment by trhumphries** [comment]
Has this ever been addressed?
